epinar/assignment2/ancestors.py

```diff
@@ -0,0 +1,113 @@
+class TreeNode:
+	def __init__(self, data):
+		self.data = data
+		self.left = None
+		self.right = None
+
+# Initiates a tree with an input value. Keeps the nodes in a dictionary for fast reach.
+# Each data(key) value should be unique.
+class BinaryTree:
+	def __init__(self, data):
+		self.tree = {}
+		node = TreeNode(data)
+		self.head = node
+		self.tree[data] = node
+		self.a =[] # shared variable between classes
+	# We assume that parent node exists and the right/left child is free.
+	def add_left(self, parent, data):
+		p = self.tree[parent]
+		node = TreeNode(data)
+		p.left = node
+		self.tree[data] = node
+	def add_right(self, parent, data):
+		p = self.tree[parent]
+		node = TreeNode(data)
+		p.right = node
+		self.tree[data] = node
+
+	# Prints in dictionary order.
+	def print_dict(self):
```

Comment:
Please use the conventional name for this method: __repr__.
This simplifies the tasks for the user: to get a reasonable representation, they can simply get it with the builtin print, instead need to use: 
>>> print my_node.print()
>>> print my_node

When we use several dozens Python modules in a project, if we do not have to learn each of their peculiar methods names, we are more productive.

Reply:
Totally makes sense, I will consider it next time!